llama-index-integrations/embeddings/llama-index-embeddings-vertex/tests/test_embeddings_vertex.py

```diff
@@ -0,0 +1,258 @@
+        self.assertEqual(positional_args[0][0].task_type, "RETRIEVAL_QUERY")
+        self.assertEqual(result, [0.1, 0.2, 0.3])
+        self.assertTrue(keyword_args["auto_truncate"])
+
+
+class VertexMultiModalEmbeddingTest(unittest.TestCase):
+    @patch("vertexai.init")
+    def test_init(self, mock_init: Mock):
+        mock_cred = Mock(return_value="mock_credentials_instance")
+        embedding = VertexMultiModalEmbedding(
+            model_name="multimodalembedding",
+            project="test-project",
+            location="us-test-location",
+            credentials=mock_cred,
+            embed_dimension=1408,
+            embed_batch_size=100,
+        )
+
+        mock_init.assert_called_once_with(
+            project="test-project",
+            location="us-test-location",
+            credentials=mock_cred,
+        )
+
+        self.assertEqual(embedding.model_name, "multimodalembedding")
+        self.assertEqual(embedding.embed_batch_size, 100)
+        self.assertEqual(embedding.embed_dimension, 1408)
+
+    @patch("vertexai.init")
+    @patch("vertexai.vision_models.MultiModalEmbeddingModel.from_pretrained")
+    def test_text_embedding(self, model_mock: Mock, init_mock: Mock):
+        model = MagicMock()
+        model_mock.return_value = model
+
+        embedding = VertexMultiModalEmbedding(
+            project="test-project",
+            location="us-test-location",
+            embed_dimension=1408,
+            additional_kwargs={"additional_kwarg": True},
+        )
+
+        model.get_embeddings.return_value = MultiModalEmbeddingResponse(
+            _prediction_response=None, text_embedding=[0.1, 0.2, 0.3]
+        )
+
+        result = embedding.get_text_embedding("some text")
+        self.assertEqual(result, [0.1, 0.2, 0.3])
+
+        model.get_embeddings.assert_called_once()
+        positional_args, keyword_args = model.get_embeddings.call_args
+
+        self.assertEqual(keyword_args["contextual_text"], "some text")
+        self.assertEqual(keyword_args["dimension"], 1408)
+        self.assertTrue(keyword_args["additional_kwarg"])
+
+    @patch("vertexai.init")
+    @patch("vertexai.vision_models.Image.load_from_file")
+    @patch("vertexai.vision_models.MultiModalEmbeddingModel.from_pretrained")
+    def test_image_embedding_path(
+        self, model_mock: Mock, load_file_mock: Mock, init_mock: Mock
+    ):
+        model = MagicMock()
+        model_mock.return_value = model
+
+        embedding = VertexMultiModalEmbedding(
+            project="test-project",
+            location="us-test-location",
+            embed_dimension=1408,
+            additional_kwargs={"additional_kwarg": True},
+        )
+
+        model.get_embeddings.return_value = MultiModalEmbeddingResponse(
+            _prediction_response=None, image_embedding=[0.1, 0.2, 0.3]
+        )
+
+        result = embedding.get_image_embedding("data/test-image.jpg")
```

Comment:
is it possible to just create a blank image on the fly for testing?

Reply:
Yeah that seems like a good idea.

Reply:
Changed for `test_image_embedding_bytes` so that it creates an empty image, and modified this to assert underlying call with file path, incase of file perm for CI.